### PR TITLE
MNT: Move cxi subdirectory to separate repo

### PR DESCRIPTION
**Pull request by ZryletTC**
First step of splitting this repo into multiple smaller repos. For now, I'm just gitignoring the repos that have already been moved over but eventually this whole repo should probably disappear.

The new cxi repo is here: https://github.com/pcdshub/cxi-config
When this PR is merged, I'll pull into /cds/group/pcds/config then clone that repo as cxi.

If all goes well, this process will be repeated for all other hutches as well (and probably the non-hutches too?)

Currently testing...
I was concerned this might disrupt iocmanager's commits but it looks like it doesn't do any git pushes so it shouldn't care.
Can you think of any other tools that might depend on the current structure?

Also open to suggestions/criticisms on procedure here. I looked into subtrees and submodules and decided those wouldn't work as nicely for this particular use case, but I might be convinced otherwise if you have a good argument.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,6 @@ iocmanager.json
 svn
 *.save
 *.sav
+
+# Folders moved to their own repos:
+cxi
```
Comment (klauer) [inline .gitignore:17]:
```suggestion
/cxi
```

**Review by klauer — APPROVED** [review]
It's sensible to split these out.

A couple thoughts:
* I think it'd be wise to name them a bit more specifically. `epics-config-cxi`, perhaps?
* It'd be a good idea to document the repository splitting effort you did (retaining the history) somewhere on confluence or even this repo.
* At some point, switching this repository to be an index of all of those separate repositories as submodules might be nice

**Comment by ZryletTC** [comment]
> * I think it'd be wise to name them a bit more specifically. `epics-config-cxi`, perhaps?

I'm open to better naming but I never understood why this repo was called `epics-config`. It seems to have very little to do with epics...

> * It'd be a good idea to document the repository splitting effort you did (retaining the history) somewhere on confluence or even this repo.

Yeah, I can write something if you'd like but it's pretty much just this: https://coderwall.com/p/a3a5xg/splitting-a-project-sub-directory-to-a-new-git-repo

> * At some point, switching this repository to be an index of all of those separate repositories as submodules might be nice. It's not strictly required, but it would make for an easier way to browse on GitHub, and make it easier for tangentially related projects like [ioc-machine-core](https://github.com/pcdshub/ioc-machine-core/blob/2c178435acd9ad3c66cd1ff44e2069f137887a3d/.gitmodules#L10-L12) to use every configuration

I like the idea of using submodules to link to the new repos. The only reason I didn't go with it in the first place is that it points to a specific commit of the submodule and I thought that would be painful to deal with. I guess if nobody's checking the status/diffs anyway, it won't matter... 🤔

**Comment by klauer** [comment]
> I'm open to better naming but I never understood why this repo was called `epics-config`. It seems to have very little to do with epics...

That's fair - though I think _most_ of it has to do with EPICS (IOC config, experimental state tracker, camera tools, parameter manager, ...).
I'd guess the naming here was likely because "it's not the hutch-python config" which is using a more nondescript name: https://github.com/pcdshub/cxi/
In retrospect, perhaps `cxi` should have been `hutch-python-(config-)cxi`.

> Yeah, I can write something if you'd like but it's pretty much just this: https://coderwall.com/p/a3a5xg/splitting-a-project-sub-directory-to-a-new-git-repo

If we're not going to split out all of them at once, it'd be good to at least link it.

> I like the idea of using submodules to link to the new repos. The only reason I didn't go with it in the first place is that it points to a specific commit of the submodule and I thought that would be painful to deal with. I guess if nobody's checking the status/diffs anyway, it won't matter... 🤔

It won't be ideal because (as you say) it points to specific commits.  That said, there's a single command that you can use to update all submodules pretty easily, at least. The aggregation would be more useful for us than it would be for any specific hutch.

**Comment by ZryletTC** [comment]
Changing approach to using submodules. Closing in favor of #2

**Pull request closed without merging**